omod/src/main/java/org/openmrs/module/appointments/web/controller/AppointmentController.java

```diff
@@ -191,4 +192,24 @@ public ResponseEntity<Object> rescheduleAppointment(@PathVariable("uuid") String
         }
     }
 
+    /**
+     * Returns a list of appointment on a date and status
+     * @param forDate the appointment date
+     * @param status - the appointment status i.e. scheduled, cancelled, completed,
+     * @return list
+     */
+    @RequestMapping(method = RequestMethod.GET, value = "appointmentStatus")
+    @ResponseBody
+    public ResponseEntity<Object> getAppointmentByDateAndStatus(@RequestParam(value = "forDate") String forDate, @RequestParam(value = "status") String status)  {
+        try {
+            if(StringUtils.isEmpty(forDate) || StringUtils.isEmpty(status)) {
+                return new ResponseEntity<>("The request requires appointment date and status", HttpStatus.BAD_REQUEST);
+            }
```

Comment:
This is where you should validate both the date (parse it into a Date object) and the status (parse it into the enum).  If either fail, send an error response.  Then pass the date and the actual enum into the service.

Reply:
So one reason I think for passing in a string here instead of an enum is that, looking at the larger PR that was originally issued (https://github.com/Bahmni/openmrs-module-appointments/pull/121/files#diff-65f85cb43e49116cee3dce7e2e006a06cbaf9d83de10b6d4af933eab4ca75853R209) the intent is to allow searching on "statuses" that aren't part of the Status enum ("Pending" and "Honoured")... these states aren't defined by actual statuses stored on the appointment, but rather are defined by custom DAO methods, "getPendingAppointments", "getHonouredAppointments".  

I'm actually wondering if we actually need/want to add a getAppointmentByDateAndStatus endpoint and service method at all, or if  rather we should make custom endpoints for "pendingAppointments" and "honourAppoinments" like we have for "earlyAppointments" and "unScheduledAppointments"?  

Making "getAppointmentByDateAndStatus" search on both "real" enum statuses and derived "statuses" seems confused...

Reply:
So my above comment is a bit tangential to this... but, yes, if we want to merge this commit in as-is, as @mseaton said this is where we should parse into an enum,